bin/dart

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# Copyright 2014 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# ---------------------------------- NOTE ---------------------------------- #
+#
+# Please keep the logic in this file consistent with the logic in the
+# `dart.bat` script in the same directory to ensure that Flutter & Dart continues
+# to work across all platforms!
+#
+# -------------------------------------------------------------------------- #
+
+set -e
+
+unset CDPATH
+
+function follow_links() {
+  cd -P "${1%/*}"
+  local file="$PWD/${1##*/}"
+  while [[ -h "$file" ]]; do
+    # On Mac OS, readlink -f doesn't work.
+    cd -P "${file%/*}"
+    file="$(readlink "$file")"
+    cd -P "${file%/*}"
+    file="$PWD/${file##*/}"
+  done
+  echo "$PWD/${file##*/}"
+}
+
+# Convert a filesystem path to a format usable by Dart's URI parser.
+function path_uri() {
+  # Reduce multiple leading slashes to a single slash.
```

Comment:
What about multiple embedded slashes? e.g. "/foo//bar//baz".
Also, what about things like "/foo/../bar/./baz"?

Actually, this looks like a bug in follow_links. Why don't you just fix follow_links to not emit double-leading slashes if $PWD is "/", which should be the only time it will emit a double slash.